### ISSUE TITLE
Iron doesn't fulfill requests under performance testing

### ISSUE THREAD

**Issue by Jonh000001** [iron/iron#568]
Hello 
I'm running the "hello world" example from Iron under a virtual SSD single-core machine in the cloud, testing with `wrk --latency -t2 -c100 -d60s` on a different single-core virtual machine from the same data center.
I get approximately 10000 - 15000 requests/s
When I try to open the hello world page in my browser from my machine I only get a response from the server after 60 seconds, when the test is done.

Is there a way to fix it?

**Iron version:** 0.6.0

**Code**
Compiled in Release mode on the latest Manjaro
```
extern crate iron;
use iron::prelude::*;
use iron::status;

fn main() {
    Iron::new(|_: &mut Request| {
        Ok(Response::with((status::Ok, "Hello World!")))
    }).http("0.0.0.0:3000").unwrap();
}
```
**Setup**

My PC -> Rust backend in the cloud on SSD VPS (Ubuntu 17.10 amd64)
`wrk --latency -t8 -c100 -d10s http://server:3000`
`Requests/sec:    146.30`

Simultaneously from another VM as the client in the same data center.
VM (Ubuntu 17.10 amd64) -> Server Nginx -> backed on Rust (hello world)
`wrk --latency -t2 -c100 -d10s http://server`
`Requests/sec:    255.88`

**Running wrk on my VM in the cloud **alone****  
VM -> backend
`wrk --latency -t2 -c100 -d10s http://server:3000`
`Requests/sec:  24816.01`

**Simultaneously running, again.**
VM -> backend
`wrk --latency -t2 -c100 -d10s http://server:3000`
`Requests/sec:  24816.01`

PC -> nginx -> backend
`wrk --latency -t8 -c100 -d10s http://server`
`Requests/sec:     49.91`

***Upd**
Simultaneously VM -> nginx and My PC > nginx
VM 
`Requests/sec: 20266.25`
PC 
`Requests/sec: 1405.94`

**Comment by kamyuentse** [comment]
The performance depends on your **network quality**. Maybe the best way to fix it is to link your PC to datacenter via fiber directly.

**Comment by kamyuentse** [comment]
Because iron based on hyper 0.10.x currently, which use sync IO.

**Comment by phlmn** [comment]
This works perfectly on my machine. Maybe this was fixed by #523 ?
